packages/amplify_datastore/lib/method_channel_datastore.dart

```diff
@@ -1,172 +1,208 @@
-    } else {
-      // This shouldn't happen. All exceptions coming from platform for
-      // amplify_datastore should have a known code. Throw an unknown error.
-      return DataStoreException(
-          AmplifyExceptionMessages.missingExceptionMessage,
-          recoverySuggestion:
-              AmplifyExceptionMessages.missingRecoverySuggestion,
-          underlyingException: e.toString());
-    }
-  }
-}
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_core/types/index.dart';
+import 'package:amplify_datastore/amplify_datastore.dart';
+import 'package:flutter/services.dart';
+import 'package:amplify_core/types/exception/AmplifyExceptionMessages.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+
+const MethodChannel _channel = MethodChannel('com.amazonaws.amplify/datastore');
+
+/// An implementation of [AmplifyDataStore] that uses method channels.
+class AmplifyDataStoreMethodChannel extends AmplifyDataStore {
+  var _allModelsStreamFromMethodChannel = null;
+
+  /// This method adds model schemas which is necessary to instantiate native plugins
+  /// This is needed before the Amplify.configure() can be called, since the native
+  /// plugins are needed to be added before that. As this function is marked for
+  /// deprecation, it actually now invokes configureDataStore internally.
+  @deprecated
+  @override
+  Future<void> configureModelProvider(
+      {ModelProviderInterface modelProvider}) async {
+    try {
+      return await _channel
+          .invokeMethod('configureDataStore', <String, dynamic>{
+        'modelSchemas':
+            modelProvider.modelSchemas.map((schema) => schema.toMap()).toList(),
+        'modelProviderVersion': modelProvider.version
+      });
+    } on PlatformException catch (e) {
+      if (e.code == "AmplifyAlreadyConfiguredException") {
+        throw AmplifyAlreadyConfiguredException(
+            AmplifyExceptionMessages.alreadyConfiguredDefaultMessage,
+            recoverySuggestion:
+                AmplifyExceptionMessages.alreadyConfiguredDefaultSuggestion);
+      } else {
+        throw _deserializeException(e);
+      }
+    }
+  }
+
+  /// This method instantiates the native DataStore plugins with plugin
+  /// configurations. This needs to happen before Amplify.configure() can be
+  /// called.
+  @override
+  Future<void> configureDataStore(
```

Comment:
This file is rewritten wholly as it was in CRLF mode, covered it into LF mode to align with other files in the codebase. The actual changes in this file include deprecating `configureModelProvider` method, introducing `configureDataStore`